test/integration/scheduler/predicates_test.go

```diff
@@ -1106,7 +1106,49 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 				return nil
 			},
 		},
-		// TODO(#91111): Add more test cases.
+		{
+			name: "other pod gets added",
+			init: func(cs kubernetes.Interface, _ string) error {
+				node, err := createNode(cs, "node-1", nil)
+				if err != nil {
+					return fmt.Errorf("cannot create node: %v", err)
+				}
+				if err := utils.AddLabelsToNode(cs, node.Name, map[string]string{"region": "test"}); err != nil {
+					return fmt.Errorf("cannot add labels to node: %v", err)
+				}
```

Comment:
Cannot we just initialize the node with the labels?

Reply:
I was replicating the same logic in this other test: https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler/predicates_test.go#L47-L58.

I could not find a test utility to create nodes with labels, do you know if there is any? If not I am not sure it's worth to add, but what do you think?

Reply:
Sigh... we should really be cleaning existing mistakes, otherwise, other developers will copy them over and over.

For now, please leave a TODO here, as well as https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler/predicates_test.go#L47-L58. A good and consistent way to craft nodes/pods is to use pkg/scheduler/testing/wrapper.go - but it's a big item.

Reply:
Will do, I am happy to work on the cleanup after this PR.